### DISCUSSION TITLE
Serial MIDI: H11 getting hot

### DISCUSSION THREAD

**Discussion by beeltink** [probonopd/MiniDexed#489] [Q&A]
I made the simple MIDI IN on a small PCB, using the H11 optocoupler.
It worked fine until this week (at least I thought it did).
I noticed MIDI IN was not working and I also noticed the H11 was getting awfully hot, like it's draining a lot of current.
I measured the voltages on the pins:
PIN6: (VDD): 3.3V, connected to PIN1 of the RPI
PIN5: (GND): 0V, connected to PIN9 of the RPI
PIN4: (OUT): 3.3V, connected to PIN10 of the RPI

When I use USB-MIDI, all works as expected.

I'm totally puzzled and wonder what may have caused this.

Schematic:

R1 = 1k
R2 = 220R
C1 = 100n
D1 = 1N4148

EDIT:
Okay, the solution to the problem: it's the idiot typing this...
In a rush, I didn't buy the H11L1, but the H11A2...
This doesn't seem to be a big deal, but the L1 has a logical output (Schmitt trigger) and the A2 has a transistor output. And, when connecting the transistor this way, full current flows from RPi pin1 through the transistor to RPi pin 10...
I guess I'm lucky I didn't damage the RPi.

![image](https://user-images.githubusercontent.com/93625613/236715538-a73d1c4d-f344-43da-a342-8ab5ffe1b3ee.png)

H11L1:
![image](https://github.com/probonopd/MiniDexed/assets/93625613/07912680-94e9-4ccf-a245-c0cf3ff7c4e6)

H11A2:
![image](https://github.com/probonopd/MiniDexed/assets/93625613/ac07108b-0eaa-4ab3-8d19-4a591a2652bc)

**Comment by Banana71** [comment]
Hi Bart, glad to hear that you find the problem. I was stumped.

**Reply by beeltink** [reply]
Did you have the same problem?

**Comment by diyelectromusic** [comment]
Well done for spotting the issue!  I must admit I'm not entirely sure how this is vastly different to using a 6N138 based circuit though, which presumably has paths through the pair of transistors from VCC to GND with no additional resistors... or is the issue that the collector is at GND so the transistor is "reverse active"...?

But I'm not really an electronics person, so don't know what I'm talking about now, so need to shut up! 
But I'd like to know the electronics explanation and why a 6N138 doesn't need extra resistors :)

Kevin

**Reply by beeltink** [reply]
I'd have to see the entire circuit.
The 6N138 looks like this, similar to the H11A2:
![image](https://github.com/probonopd/MiniDexed/assets/93625613/1d5bbc44-d62a-4aec-a361-9876afc892ee)

The 6N137 is more like the H11L1:
![image](https://github.com/probonopd/MiniDexed/assets/93625613/d69bd548-b921-48c4-a0d8-2455f562add7)

**Reply by diyelectromusic** [reply]
I was just thinking of the MIDI circuit that seems to be everywhere if one Googles 6N138 MIDI :)  But I guess the most used one is probably the one that accompanies the Notes and Volts MIDI tutorial for Arduino: https://www.notesandvolts.com/2015/02/midi-and-arduino-build-midi-input.html

Kevin

**Reply by diyelectromusic** [reply]
Ok, watching that again, I can see how the current flows through the pull-up resistor when the transistor is active, so current is still limited, bringing RX low.

So the question now is why does your version overheat, or as I say is it that due to the transistor being reversed?

Kevin

**Reply by beeltink** [reply]
Correct.
The H11Ax would have the base and emitter connected to +3.3V and the collector to 0V.
Also, there's no resistor to throttle the current flow.

**Reply by beeltink** [reply]
So, the H11L1 arrived today.
I orderd PDIP6 packages, but instead I received SMD... *sigh*
So, what would McGyver do?

![H11L1](https://github.com/probonopd/MiniDexed/assets/93625613/ab35b2c5-a4c4-4251-8766-894c2edc45e1)
+
![PCB](https://github.com/probonopd/MiniDexed/assets/93625613/c38625e2-c975-4c4e-b425-19479f82192a)
=
![Solution](https://github.com/probonopd/MiniDexed/assets/93625613/b9bb68b9-79c3-4af0-8625-2777c522ec80)

**Comment by Banana71** [comment]
another brick in the wall
Well done
![Download](https://github.com/probonopd/MiniDexed/assets/104296205/0056692f-a430-4c39-849d-8df49b54c32b)

**Comment by Gustavomurta** [comment]
I realized that this circuit generates many doubts. For clarification, I made these diagrams.

https://github.com/probonopd/MiniDexed/discussions/518#discussioncomment-6494743

![miniDexed MIDIN edited2](https://github.com/probonopd/MiniDexed/assets/4587366/e57c59c8-aeae-4ec7-bd43-aecea8a68a42)

**Reply by nor50** [reply]
Do I need a 100nF capacitor?

**Reply by diyelectromusic** [reply]
I think it's optional, but I'm not really an electronics person :)

Kevin

**Reply by Gustavomurta** [reply]
No capacitor needed

**Reply by nor50** [reply]
Me too, Kevin.
I needed a TRS-MIDI interface, so I started working on electronics two weeks ago.
Thanks for all the tips.

**Comment by nor50** [comment]
Thank you for your helpful post.
I also had the same experience using TLP371.
Recently I've been using a 4PIN opto-isolator like the TLP785. The circuit has become smaller.

**Reply by nor50** [reply]
https://akizukidenshi.com/goodsaffix/MIDI-IF-For-IchigoJam_rev1.pdf
I used this as a reference when making it, but it seems that the capacitor wasn't even necessary in the first place.
By the way, "Ichigo" means strawberry in Japanese.

**Reply by diyelectromusic** [reply]
The use of the capacitor (as I understand it) is simply for smoothing/evening out purposes, so not required for operation of the circuit, but possibly useful if you're running off a noisy microcontroller power line, that's all.

As I say, I'm not an electronics person, so don't do any proper analysis myself, I just tend to go with a 100nF close to the power rails of any chips or devices I'm using in my circuits that are driven by a microcontroller.  It's the kind of thing where if they aren't use and you get reliability issues, it will be very, very hard to debug, so I use them to remove the possibility in the first place.

Also, because their use is so common in this way, a schematic in a datasheet often doesn't even bother to mention them - they are just assumed to be part of the "how you build circuits" knowledge that people have - like the use of ground zones and stuff like that (again, as I understand things).

Kevin

**Comment by craigyjp** [comment]
It is for decoupling purposes, when the opto isolator switches with midi
signal then the decoupling can absorb the shock so to speak on the power
rails. It's good practice, but not 100% you must do and certainly won't
prevent the chip getting hot, that's another issue entirely.
